### ISSUE TITLE
[1.3-beta.11] Cannot import modules from packages directory inside meteor shell

### ISSUE THREAD

**Issue by runarberg** [meteor/meteor#6271]
If I create a package in my app, I can import it using `import {name} from "meteor/my-package"`. However, that seems not to be working inside `meteor shell`.

```
$ meteor new test
$ cd test
$ meteor update --release 1.3-beta.11
$ meteor add ecmascript
$ meteor create --package my-package
$ meteor run &
[1] 35695
[[[[[ /private/tmp/test ]]]]]

=> Started proxy.
=> Started MongoDB.
=> Started your app.

=> App running at: http://localhost:3000/

$ meteor shell
Welcome to the server-side interactive shell!

Tab completion is enabled for global variables.

Type .reload to restart the server and the shell.
Type .exit to disconnect from the server and leave the shell.
Type .help for additional help.

> import {name} from "meteor/my-package"
Error: Cannot find module 'meteor/my-package'
    at Function.Module._resolveFilename (module.js:338:15)
    at Function.Module._load (module.js:280:25)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at [object Object]:1:-44
    at Script.(anonymous function) [as runInThisContext] (vm.js:41:22)
    at /private/tmp/test/.meteor/local/build/programs/server/shell-server.js:363:27
    at /Users/runarberg/.meteor/packages/meteor-tool/.1.1.13-beta.11.1g8flc0++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/dev_bundle/server-lib/node_modules/meteor-promise/fiber_pool.js:33:40
```

on Mac OS X 10.10.5

**Comment by lsunsi** [comment]
Maybe this is intended behavior.
Also waiting for some info

**Comment by zol** [comment]
@benjamn - Please confirm whether this should or shouldn't work.

**Comment by benjamn** [comment]
The easy part of the answer is that your setup steps need some slight tweaking:
```sh
$ meteor create test
$ cd test
$ meteor update --release 1.3-beta.11
$ meteor add ecmascript
$ mkdir packages
$ cd packages
$ meteor create --package my-package
$ cd ..
$ echo my-package >> .meteor/packages
$ meteor run &
$ meteor shell
```
The harder part is that `import` statements are not expressions, so what's executed on the server side is merely
```js
var _myPackage = require("meteor/my-package");
```
when what you really want is to declare a new `name` variable that refers to `_myPackage.name`. Without knowing how the imported symbols are used, the compiler has no way of rewriting references in later shell commands.

Perhaps the easiest workaround for now is to call `require("meteor/my-package")`, which returns an object that's easier to use:
```js
> require("meteor/my-package")
{ __esModule: true, name: 'my-package' }
```
I realize that's not ideal, but hopefully `require` will be sufficient for your shell debugging needs.

**Comment by benjamn** [comment]
Partially fixed in [`beta.12`](https://github.com/meteor/meteor/issues/6266#issuecomment-190789852). Specifically, `require` works properly now, even if `import` statements still suffer from the problems described above.

**Comment by abernix** [comment]
This is now available in 1.3.3-beta.0.  If you want to update your project to try it, run `meteor update --release 1.3.3-beta.0`.  Please report back if it works for you!
